contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -215,63 +245,87 @@ std::string cleanPrimVarName( const std::string &primVarName )
-		m_converted = true;
+		m_primitive->variables[cleanedPrimvarName] = IECore::PrimitiveVariable( interpolation, p, indices );
 	}
 
 	IECore::PrimitivePtr m_primitive;
 	const pxr::UsdGeomPrimvar &m_primVar;
 	pxr::UsdTimeCode m_time;
-	bool m_converted;
 };
 
-void convertPrimVar( IECore::PrimitivePtr primitive, pxr::UsdTimeCode time, const pxr::UsdGeomPrimvar &primVar )
+namespace
 {
-	IECore::PrimitiveVariable::Interpolation interpolation = convertInterpolation( primVar.GetInterpolation() );
-	if( interpolation == IECore::PrimitiveVariable::Invalid )
+static std::map<pxr::TfToken, std::function<void( PrimVarConverter& converter )> > primvarConversions =
+{
+	{ pxr::TfToken( "bool[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<bool, bool, IECore::TypedData> >(); } },
+	{ pxr::TfToken( "half[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<half, pxr::GfHalf, IECore::TypedData > >(); } },
+	{ pxr::TfToken( "double[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<double, double, IECore::TypedData > >();  } },
+	{ pxr::TfToken( "int[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<int, int, IECore::TypedData> >(); } },
+	{ pxr::TfToken( "uint[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<unsigned int, unsigned int, IECore::TypedData> >(); } },
+	{ pxr::TfToken( "char[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<char, char, IECore::TypedData> >(); } },
+	{ pxr::TfToken( "float[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<float, float, IECore::TypedData> >(); } },
+	{ pxr::TfToken( "color3f[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<Imath::V3f, pxr::GfVec3f, IECore::GeometricTypedData> >(); } },
+	{ pxr::TfToken( "float3[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<Imath::V3f, pxr::GfVec3f, IECore::GeometricTypedData> >(); } },
+	{ pxr::TfToken( "float2[]" ), []( PrimVarConverter& converter ) { converter.doConversion<TypedConverter<Imath::V2f, pxr::GfVec2f, IECore::GeometricTypedData> >(); } }
+};
+
+}
+
+void convertPrimVar( IECore::PrimitivePtr primitive, const pxr::UsdGeomPrimvar &primVar, pxr::UsdTimeCode time )
+{
+	pxr::TfToken typeToken = primVar.GetTypeName().GetAsToken();
+
+	auto it = primvarConversions.find( typeToken );
+	if ( it != primvarConversions.end() )
+	{
+		PrimVarConverter converter( primitive, primVar, time );
+		it->second( converter );
+	}
+	else
 	{
-		return;
+		IECore::msg(IECore::MessageHandler::Level::Warning, "USDScene", boost::format("Unknown type %1% on PrimVar %2% ") % typeToken.GetString() % primVar.GetName().GetString());
 	}
 
-	PrimVarConverter primVarConverter( primitive, primVar, time );
-	mpl::for_each<TypeMapping>( primVarConverter );
```

Comment:
Looks like the TypeMapping and the mpl includes are unused now, so let's remove them.